plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/binance/v4/BinanceSortedGroupV4.java

```diff
@@ -0,0 +1,281 @@
+        List<BinanceBeanV4> result = new ArrayList<>();
+        if (rows.size() > 0) {
+            var time = rows.values().stream().collect(Collectors.toList()).get(0).get(0).getDate();
+            for (Map.Entry<Currency, List<BinanceBeanV4>> entry : rows.entrySet()) {
+                var newBean = new BinanceBeanV4();
+                var currency = entry.getKey();
+                var change = entry.getValue().stream().map(BinanceBeanV4::getChange).reduce(BigDecimal.ZERO, BigDecimal::add);
+                var ids = entry.getValue().stream().map(BinanceBeanV4::getRowId).collect(Collectors.toList());
+                if (entry.getValue().size() > 0) {
+                    newBean.setChange(change);
+                    newBean.setCoin(currency);
+                    newBean.usedIds.addAll(ids);
+                    newBean.setDate(time);
+                    result.add(newBean);
+                }
+            }
+        }
+        return result;
+    }
+
+    private void validateBuySell() {
+        if (rowBuySellRelated.size() != 2) {
+            throw new DataIgnoredException("Wrong number of currencies");
+        }
+        // one of them must be plus and second minus
+        var stValue = rowBuySellRelated.get(0).getChange();
+        var ndValue = rowBuySellRelated.get(1).getChange();
+        var minus = stValue.multiply(ndValue); // must be everytime minus
+        if (minus.compareTo(BigDecimal.ZERO) > 0) {
+            throw new DataIgnoredException("Wrong change value");
+        }
+    }
+
+    private void validateFee() {
+        if (rowFees.size() > 0 && !(rowBuySellRelated.size() > 0 || rowDeposit.size() > 0 || rowWithdrawal.size() > 0)) {
+            throw new DataIgnoredException("No txs for fee");
+        }
+    }
+
+    private boolean isCrypto(List<BinanceBeanV4> rows) {
+        boolean result = false;
+        for (BinanceBeanV4 row : rows) {
+            if (!row.getCoin().isFiat()) {
+                result = true;
+            }
+        }
+        return result;
+    }
+
+    public void createTransactions() {
+        int buySellNum = rowBuySellRelated.size();
+        int depositNum = rowDeposit.size();
+        int withdrawNum = rowWithdrawal.size();
+        int feeNum = rowFees.size();
+
+        if (buySellNum > 0) {
+            validateBuySell();
+            createBuySellTxs();
+        }
+
+        if (depositNum > 0 || withdrawNum > 0) {
+            createDepositWithdrawalTxs();
+        }
+
+        if (feeNum > 0) {
+            validateFee();
+            addFeeToTxs();
+        }
+        createdTransactions.stream().forEach(r -> {
+            r.setInTransaction(true);
+        });
+    }
+
+    private void addFeeToTxs() {
+        try {
+            var txs = createdTransactions.get(0);
+            for (BinanceBeanV4 fee : rowFees) {
+                fee.setInTransaction(true);
+                var bean = new BinanceBeanV4();
```

Comment:
Better would be if `BinanceBeanV4` had `@Builder` lombok annotation and here you would use builder pattern. Something like : 

```
BinanceBeanV4.builder()
    .type(FEE)
    .fee(...)
    .......
    .build()
```